api/src/main/java/org/apache/iceberg/expressions/ExpressionVisitors.java

```diff
@@ -55,6 +55,14 @@ public <T> R predicate(BoundPredicate<T> pred) {
     public <T> R predicate(UnboundPredicate<T> pred) {
       return null;
     }
+
+    public <T> R aggregate(BoundAggregate<T> agg) {
+      return null;
```

Comment:
Is it safe to return `null` for all visitors that may have an `Aggregate` passed in through `Expression`?

I think we may want to throw `UnsupportedOperationException` to ensure that aggregates are either explicitly supported or will be rejected.